lightning/src/chain/channelmonitor.rs

```diff
@@ -2121,7 +2131,8 @@ macro_rules! fail_unbroadcast_htlcs {
 									commitment_tx_output_idx: None,
 								},
 							};
-							log_trace!($logger, "Failing HTLC with payment_hash {} from {} counterparty commitment tx due to broadcast of {} commitment transaction {}, waiting for confirmation (at height {})",
+							let logger = &WithContext::from_channel_monitor_impl($logger, $self);
```

Comment:
Why take a reference here and elsewhere?

Reply:
this actually goes back to some Logger instances being `L` and some `&L` where `L: Deref` and `L::Target: Logger`. That causes sometimes the need for an extra layer of deref to get to the object. from_** actually takes  

`WithContext:: from_channel_monitor_impl` here returns `WithContext` which is a `Logger` but doesnt fit the `&L` signature for some methods. hence the reference is needed.

Reply:
But that should only be an issue if you need to pass the logger to a function taking `&L`. Here, it's passed directly to the logging macro.